[DOC] Update dapp-frontend.mdx so it includes how to use the StellarWalletsKit to connect the wallets (#913)
stellar/stellar-docs: docs/build/apps/dapp-frontend.mdx

Now you're all set up to use Freighter as a user, and you can add it to your app.

### Add the StellarWalletsKit and set it up

> Even doe we're using Freighter to test our app, there are more wallets that support signing soroban transactions and to make their integration easier we are using the `StellarWalletsKit` library which allows us supporting all Stellar Wallets with a single library.

Reviewer: ```suggestion
Even though we're using Freighter to test our app, there are more wallets that support signing smart contract transactions. To make their integration easier, we are using the `StellarWalletsKit` library which allows us support all Stellar Wallets with a single library.
```

To install this kit we are going to include the next package:

```shell
npm install @creit.tech/stellar-wallets-kit
```

With the package installed, we are going to create a new simple file where our instantiated kit and simple state will be located. Create the file `src/stellar-wallets-kit.ts` and paste this:

```ts title="src/stellar-wallets-kit.ts"
import {
  allowAllModules,
  FREIGHTER_ID,
  StellarWalletsKit,
  WalletNetwork,
} from "@creit.tech/stellar-wallets-kit";

const kit: StellarWalletsKit = new StellarWalletsKit({
  modules: allowAllModules(),
  network: WalletNetwork.TESTNET,
  selectedWalletId: FREIGHTER_ID,
});

const connectionState: { publicKey: string | undefined } = {
  publicKey: undefined,
};

function loadedPublicKey(): string | undefined {
  return connectionState.publicKey;
}

function setPublicKey(data: string): void {
  connectionState.publicKey = data;
}

export { kit, loadedPublicKey, setPublicKey };
```

> In the code above we created an instance of the kit and two simple functions that will take care of "setting" and "loading" the public key of the user in case we want to use it elsewhere in our code. The kit is started with Freighter as the default wallet and the Testnet network as the default network, you can check all the details about how the kit works at [https://stellarwalletskit.dev/](https://stellarwalletskit.dev/)

Reviewer: ```suggestion
In the code above, we created an instance of the kit and two simple functions that will take care of "setting" and "loading" the public key of the user. This lets us use the user's public key elsewhere in our code. The kit is started with Freighter as the default wallet, and the Testnet network as the default network. You can learn more about how the kit works in [the StellarWalletsKit documentation](https://stellarwalletskit.dev/)
```

> Now we're going to add a "Connect" button to the page that opens the kit's built in modal and prompts the user to use their preferred wallet. Once the user picks its preferred wallet and grants permission to accept requests from the website, we will fetch the public key and the "Connect" button will be replaced with a message saying, "Signed in as [their public key]".

Reviewer: ```suggestion
Now we're going to add a "Connect" button to the page which will open the kit's built in modal, and prompt the user to use their preferred wallet. Once the user picks their preferred wallet and grants permission to accept requests from the website, we will fetch the public key and the "Connect" button will be replaced with a message saying, "Signed in as [their public key]".
```

</style>

<script>

>   import { kit, setPublicKey } from '../stellar-wallets-kit';

Reviewer: I'm not well-versed in Astro, so this might be a stupid question. In SvelteKit, there's a "shortcut" for the `/src/lib/` directory (`$lib`), in Docusaurus (react-based) there's one for the project root (`@site/`), i think Vue has one for the project root, too (`@` maybe?).

Is there a similar kind of thing for Astro? Or, are relative file paths what's typically used?

Author: @ElliotFriend to be honest this was the first time I used Astro. I used direct file paths because it was the way the Stellar guides use them and that makes me think it's not something Astro supports

Reviewer: Ah. Gotcha. Sounds solid enough to me hehe

You can read more about this in [Astro's page about client-side scripts](https://docs.astro.build/en/guides/client-side-scripts/).

> The code itself here is pretty self-explanatory. We import the kit from the file we created before, then when the user clicks on the button we launch the built-in modal and once the user picks their preferred wallet, we set it as the default one before requesting and saving the address.

Reviewer: ```suggestion
The code itself here is pretty self-explanatory. We import the wallets kit from the file we created before. Then, when the user clicks on the button, we launch the built-in modal do display to the user connection options. Once the user picks their preferred wallet, we set it as the wallets kit's default wallet before requesting and saving the address.
```
